build-support/bin/generate_github_workflows.py

```diff
@@ -293,6 +293,7 @@ def test_workflow_jobs(primary_python_version: str, *, cron: bool) -> Jobs:
             "runs-on": LINUX_VERSION,
             "needs": "bootstrap_pants_linux",
             "strategy": {"matrix": {"python-version": [primary_python_version]}},
+            "timeout-minutes": 30,
```

Comment:
Oh hm, this is too short. Should be 45 minutes. Sorry, I know this PR is frustrating.